spec/compiler/type_inference/global_spec.cr

```diff
@@ -185,6 +185,20 @@ describe "Global inference" do
       )) { types["Foo"] }
   end
 
+  it "doesn't infer from new if generic" do
+    assert_error %(
+      class Foo(T)
+        def self.new
+          10
+        end
+      end
+
+      $x = Foo.new
+      $x
+      ),
+      "undefined global variable '$x'"
```

Comment:
So maybe say "can't infer type of $x" or something? "Undefined global variable" makes me think the assignment is missing.